shared/middleware/files_api.rb

```diff
@@ -52,7 +52,7 @@ def allowed_file_type?(endpoint, extension)
   # Read a file. Optionally get a specific version instead of the most recent.
   #
   get %r{/v3/(assets|sources)/([^/]+)/([^/]+)$} do |endpoint, encrypted_channel_id, filename|
-    dont_cache
+    dont_cache unless endpoint == 'assets'
```

Comment:
Could we add tests to make sure that caching behavior is correct for assets and non-assets?